tests/pytests/functional/modules/test_cmdmod.py

```diff
@@ -0,0 +1,564 @@
+    """
+    with _ensure_user_exists(runas_usr, usermod):
+        user_path = cmdmod.run_stdout('printf %s "$PATH"', runas=runas_usr)
+    # XXX: Not sure of a better way. Environment starts out with
+    # /bin:/usr/bin and should be populated by path helper and the bash
+    # profile.
+    assert "/bin:/usr/bin" != user_path
+
+
+@pytest.mark.destructive_test
+@pytest.mark.skip_if_not_root
+@pytest.mark.skip_unless_on_darwin(reason="Applicable to MacOS only")
+@pytest.mark.slow_test
+def test_runas_complex_command_bad_cwd(cmdmod, usermod, runas_usr, tmp_path):
+    """
+    cmd.run should not accidentally run parts of a complex command when
+    given a cwd which cannot be used by the user the command is run as.
+
+    Due to the need to use `su -l` to login to another user on MacOS, we
+    cannot cd into directories that the target user themselves does not
+    have execute permission for. To an extent, this test is testing that
+    buggy behaviour, but its purpose is to ensure that the greater bug of
+    running commands after failing to cd does not occur.
+    """
+    tmp_cwd = str(tmp_path)
+    os.chmod(tmp_cwd, 0o700)
+
+    with _ensure_user_exists(runas_usr, usermod):
+        cmd_result = cmdmod.run_all(
+            'pwd; pwd; : $(echo "You have failed the test" >&2)',
+            cwd=tmp_cwd,
+            runas=runas_usr,
+        )
+
+    assert "" == cmd_result["stdout"]
+    assert "You have failed the test" not in cmd_result["stderr"]
+    assert 0 != cmd_result["retcode"]
+
+
+@SKIP_INITIAL_PHOTONOS_FAILURES
+@pytest.mark.skip_on_windows
+@pytest.mark.skip_if_not_root
+@pytest.mark.destructive_test
+@pytest.mark.slow_test
+def test_runas(cmdmod, usermod, runas_usr):
+    """
+    Ensure that the env is the runas user's
+    """
+    with _ensure_user_exists(runas_usr, usermod):
+        out = cmdmod.run("env", runas=runas_usr).splitlines()
+    assert "USER={}".format(runas_usr) in out
+
+
+@pytest.mark.skip_if_binaries_missing("sleep", reason="sleep cmd not installed")
+def test_timeout(cmdmod):
+    """
+    cmd.run trigger timeout
+    """
+    out = cmdmod.run("sleep 2 && echo hello", timeout=1, python_shell=True)
+    assert "Timed out" in out
+
+
+@pytest.mark.skip_if_binaries_missing("sleep", reason="sleep cmd not installed")
+def test_timeout_success(cmdmod):
+    """
+    cmd.run sufficient timeout to succeed
+    """
+    out = cmdmod.run("sleep 1 && echo hello", timeout=2, python_shell=True)
+    assert out == "hello"
+
+
+@pytest.mark.slow_test
+def test_cmd_run_whoami(cmdmod, running_username):
+    """
+    test return of whoami
+    """
+    if not salt.utils.platform.is_windows():
+        user = running_username
+    else:
+        user = salt.utils.user.get_specific_user()
+    if user.startswith("sudo_"):
+        user = user.replace("sudo_", "")
+    cmd = cmdmod.run("whoami")
+    assert user.lower() == cmd.lower()
+
+
+@pytest.mark.skip_unless_on_windows(reason="Minion is not Windows")
```

Comment:
```suggestion
@pytest.mark.skip_unless_on_windows
```
No need to customize message, does not add anything.